cloud/src/meta-service/meta_service_job.cpp

```diff
@@ -124,6 +136,17 @@ void start_compaction_job(MetaServiceCode& code, std::string& msg, std::stringst
     }
     while (err == TxnErrorCode::TXN_OK) {
         job_pb.ParseFromString(job_val);
+        if (job_pb.has_schema_change() && !check_compaction_input_verions(compaction, job_pb)) {
```

Comment:
why not put `job_pb.has_schema_change()` into `check_compaction_input_verions()`